### PR TITLE
[TT-4569] optional header authconfig

### PR DESCRIPTION
**Pull request by jeffy-mathew**
Add disableHeader flag to authConfig struct
Add validation to check if at least one auth source is enabled

### CODE REVIEW

File: apidef/api_definitions.go [hunk 1]
```diff
@@ -601,6 +601,7 @@ type AuthConfig struct {
 	ParamName         string          `mapstructure:"param_name" bson:"param_name" json:"param_name"`
 	UseCookie         bool            `mapstructure:"use_cookie" bson:"use_cookie" json:"use_cookie"`
 	CookieName        string          `mapstructure:"cookie_name" bson:"cookie_name" json:"cookie_name"`
+	DisableHeader     bool            `mapstructure:"use_header" bson:"use_header" json:"use_header"`
```
Comment (furkansenharputlu) [inline apidef/api_definitions.go:604]: json tag name should not be use_header

File: apidef/validator.go [hunk 2]
```diff
@@ -91,3 +93,36 @@ func (r *RuleUniqueDataSourceNames) Validate(apiDef *APIDefinition, validationRe
 		usedNames[trimmedName] = true
 	}
 }
+
+var ErrAllAuthSourcesDisabled = "all auth sources are disabled for %s, at least one of header/cookie/param must be enabled"
+
+type RuleAtLeastEnableOneAuthConfig struct{}
```
Comment (furkansenharputlu) [inline apidef/validator.go:99]: At least one auth source, auth config corresponds to another thing in our stack

File: apidef/api_definitions.go (hunk 1 above)
Comment (jeffy-mathew) [reply apidef/api_definitions.go]: Updated

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3886/merge
Commit: 8d0a0af341c5ec7f33012c500105fdc71a7eb3d2 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1829188971)

**Review by urbanishimwe — APPROVED** [review]
LGTM

**Comment by Tyk-ITS** [comment]
**API tests result: skipped** :no_entry_sign: 
Branch used: refs/pull/3886/merge
Commit: ac38d696bdd4afe9f4f6f2ad77971b5504166b0e 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1851857664)

**Review by furkansenharputlu — APPROVED** [review]

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3886/merge
Commit: 2b099ea846f590beef603f858cd828c2dc77947d 
Triggered by: pull_request (@furkansenharputlu)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1878411475)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/pull/3886/merge
Commit: 3e43df7a6b295cc85cb8813bb3b88026b241dc17 
Triggered by: pull_request (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1886235813)

**Comment by Tyk-ITS** [comment]
**API tests result: success** :white_check_mark: 
Branch used: refs/heads/master
Commit: fa71b257cbe7e15235b895d42fc47a82544c4369 [TT-4569] optional header authconfig (#3886)

* add disableHeader flag to auth config

* add authconfig validator

* fix typo

* review comment

Co-authored-by: Furkan Senharputlu <furkan_senharputlu@hotmail.com>
Triggered by: push (@jeffy-mathew)
[Execution page](https://github.com/TykTechnologies/tyk/actions/runs/1886522655)

**Pull request merged**